Sources/NIOPosix/Bootstrap.swift

```diff
@@ -1984,53 +1984,63 @@ public final class NIOPipeBootstrap {
     ///   - output: The _Unix file descriptor_ for the output (ie. the write side).
     /// - Returns: an `EventLoopFuture<Channel>` to deliver the `Channel`.
     public func takingOwnershipOfDescriptors(input: CInt, output: CInt) -> EventLoopFuture<Channel> {
-        precondition(input >= 0 && output >= 0 && input != output,
-                     "illegal file descriptor pair. The file descriptors \(input), \(output) " +
-                     "must be distinct and both positive integers.")
-        let eventLoop = group.next()
-        do {
-            try self.validateFileDescriptorIsNotAFile(input)
-            try self.validateFileDescriptorIsNotAFile(output)
-        } catch {
-            return eventLoop.makeFailedFuture(error)
-        }
+        self._takingOwnershipOfDescriptors(input: input, output: output)
+    }
 
-        let channelInitializer = self.channelInitializer ?? { _ in eventLoop.makeSucceededFuture(()) }
-        let channel: PipeChannel
-        do {
-            let inputFH = NIOFileHandle(descriptor: input)
-            let outputFH = NIOFileHandle(descriptor: output)
-            channel = try PipeChannel(eventLoop: eventLoop as! SelectableEventLoop,
-                                      inputPipe: inputFH,
-                                      outputPipe: outputFH)
-        } catch {
-            return eventLoop.makeFailedFuture(error)
-        }
+    /// Create the `PipeChannel` with the provided input file descriptors.
+    ///
+    /// The input file descriptor must be distinct.
+    ///
+    /// - Note: If this method returns a succeeded future, SwiftNIO will close `input` when the `Channel`
+    ///         becomes inactive. You _must not_ do any further operations to `input`, including `close`.
+    ///         If this method returns a failed future, you still own the file descriptor and are responsible for
+    ///         closing them.
+    ///
+    /// - Parameters:
+    ///   - input: The _Unix file descriptor_ for the input (ie. the read side).
```

Comment:
Bit confused by naming here: I expected `input` to be the input to the pipe, i.e. the outbound side of the channel and the _write_ side.

Reply:
The input here is the `read` side and that is the same for all the other methods. Do you think it is just confusing for the methods here that take a single fd?